pkg/controller/dual-pods/server-running.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2025 The llm-d Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dualpods
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/llm-d-incubation/llm-d-fast-model-actuation/pkg/api"
+	stubapi "github.com/llm-d-incubation/llm-d-fast-model-actuation/pkg/stub/api"
+)
+
+func (ctl *controller) processServerRunningPod(ctx context.Context, runningPod *corev1.Pod) (error, bool) {
+	logger := klog.FromContext(ctx)
+	logger.V(5).Info("Processing server-running pod", "name", runningPod.Name)
+
+	// get the server-requesting pod from the owner reference
+	if len(runningPod.OwnerReferences) == 0 {
+		logger.V(5).Info("No owner reference found", "name", runningPod.Name)
+		return nil, true
+	}
+	ownerRef := runningPod.OwnerReferences[0]
+	requestingPod, err := ctl.podLister.Pods(runningPod.Namespace).Get(ownerRef.Name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.V(5).Info("server-requesting pod not found", "name", runningPod.Name, "ownerName", ownerRef.Name)
+			return nil, true
```

Comment:
If the server-requesting Pod does not exist then it is time to delete the server-running Pod.